src/main/java/com/mparticle/kits/InstabotKit.java

```diff
@@ -0,0 +1,164 @@
+package com.mparticle.kits;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
+
+import com.mparticle.MPEvent;
+import com.mparticle.MParticle;
+import com.rokolabs.sdk.tools.ThreadUtils;
+
+import java.util.List;
+import java.util.Map;
+
+import io.instabot.sdk.Instabot;
+import io.instabot.sdk.Settings;
+import io.instabot.sdk.account.Account;
+import io.instabot.sdk.analytics.Analytics;
+import io.instabot.sdk.analytics.Event;
+import io.instabot.sdk.push.Push;
+import io.instabot.sdk.push.PushConstants;
+import io.instabot.sdk.push.PushNotificationHelper;
+
+public class InstabotKit extends KitIntegration implements KitIntegration.EventListener, KitIntegration.PushListener, KitIntegration.AttributeListener, InstabotProvider {
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+        if(!settings.containsKey("apiKey") || TextUtils.isEmpty(settings.get("apiKey"))){
+            Log.e("InstabotKit", "Instabot initialization was failed. Please set \"apiKey\".");
+            return null;
+        }
+
+        Instabot.start(settings.get("apiKey"), context, new Instabot.CallbackStart() {
+            @Override
+            public void start() {
+                Instabot.getSettings().setInstabotActivity("io.instabot.INSTABOT_ACTIVITY");
+            }
+        });
+        return null;
+    }
+
+    @Override
+    public Object getInstance() {
+        return this;
+    }
+
+    @Override
+    public Instabot instabot() {
+        return Instabot.getInstance();
+    }
+
+    @Override
+    public String getName() {
+        return "Instabot";
+    }
+
+    @Override
+    public List<ReportingMessage> setOptOut(boolean optedOut) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> leaveBreadcrumb(String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logError(String s, Map<String, String> map) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logException(Exception e, Map<String, String> map, String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logEvent(MPEvent mpEvent) {
+        Log.e(getName(), mpEvent.getEventName());
+        Analytics.addEvents(new Event(mpEvent.getEventName()));
+        return null;
```

Comment:
you should be returning a List<Reporting> message here. See other kits such as Appboy for an example.